woocommerce/woocommerce-ios
# Order Creation: Present New Order flow as a modal

Part of #6381
⚠️ Please make sure that this PR is merged before #6387 ⚠️ 


### Description

When working on the split view for the Orders tab in PR #6387, I found that presenting the New Order flow using the navigation stack doesn't look good on tablets. After discussing with the team (internal ref: p1646791790098809-slack-CGPNUU63E), it's decided that we should present the flow modally and disable swipe-to-dismiss to prevent accidental dismissal and loss of data.

Solution:
- Add a cancel button on the New Order button to handle manual dismissal
- Update the presentation of the New Order flow to use a non-dismissible modal.
- Also: update the presentation of Simple Payment flow to be non-dismissible too.

### Testing instructions

- Log in to a valid account
- Enable Order Creation in Settings > Experimental Features > Order Creation
- Select Orders tab > tap the (+) button and select Create new order
- Notice that the New Order screen is presented modally and cannot be dismissed by swiping down.
- Tap the Cancel button to dismiss the New Order screen. Notice that the screen is dismissed successfully.
- Tap the (+) button on the Orders screen and select Simple Payment. Notice the same behavior on the Simple Payment screen.

### Screenshots

https://user-images.githubusercontent.com/5533851/157370382-73078657-ccea-4fe5-aca8-fc9422720180.mp4

---
